alchemy/src/coinbase/evm-account.ts

```diff
@@ -0,0 +1,313 @@
+            ([network, tokens]) =>
+              tokens.map((token) => ({
+                network: network as FaucetNetwork,
+                token: token as FaucetToken,
+                key: `${network}:${token}`,
+              })),
+          );
+
+          // Filter to only combinations that are actually new
+          const toFund = newCombinations.filter(
+            ({ key }) => !oldCombinations.has(key),
+          );
+
+          if (toFund.length > 0) {
+            await Promise.all(
+              toFund.map(({ network, token }) =>
+                cdp.evm
+                  .requestFaucet({
+                    address: this.output.address,
+                    network,
+                    token,
+                  })
+                  .catch((err) => {
+                    console.warn(
+                      `⚠️ Failed to request ${token} on ${network} for ${this.output.address}: ${err.message}`,
+                    );
+                    return null;
+                  }),
+              ),
+            );
+
+            console.log(
+              `💧 Requested ${toFund.length} new faucet fund${toFund.length === 1 ? "" : "s"} for ${this.output.address}`,
+            );
+          }
+        } catch (error: any) {
+          console.warn(
+            `⚠️ Failed to request additional faucet funds for ${this.output.address}: ${error.message}`,
+          );
+          // Continue - don't break the update
+        }
+
+        return {
+          ...this.output,
+          faucet: props.faucet,
+        };
+      }
+
+      return this.output;
+    }
+
+    // Handle delete phase
+    if (this.phase === "delete") {
+      // CDP SDK doesn't support deleting accounts
+      // Accounts remain in CDP but are no longer tracked by Alchemy
+      console.log(`🗑️ Untracking EVM Account: ${this.output.address}`);
+      return {} as any;
+    }
+
+    let account;
+
+    // Handle account creation/retrieval
+    if (props.privateKey) {
+      // Import account with private key
+      // This is idempotent in CDP
+      account = await cdp.evm.importAccount({
+        privateKey: props.privateKey.unencrypted,
+        name: props.name,
+      });
+    } else {
+      // Check for adoption or use getOrCreate pattern
+      const adopt = props.adopt ?? this.scope.adopt;
+      if (adopt) {
+        // With adoption, use getOrCreate which returns existing if it exists
+        account = await cdp.evm.getOrCreateAccount({
+          name: props.name,
+        });
+      } else {
+        // Without adoption, need to check if account exists first
+        try {
+          await cdp.evm.getAccount({ name: props.name });
+          throw new Error(
+            `Account with name '${props.name}' already exists. Use adopt: true to use the existing account.`,
+          );
+        } catch (_error: any) {
+          // Account not found, create it
+          account = await cdp.evm.createAccount({ name: props.name });
+        }
```

Comment:
I think we want to just call `createAccount` in the `try` and then throw the new Error in the `catch` - no need to call `getAccount` here if I'm understanding correctly

Reply:
Good catch! Make sene, i think we could probably just rethrow origina error without doing errortype matching anyway